### DISCUSSION TITLE
Not able to curl to a service running on host machine from with in a container which part of a pod.

### DISCUSSION THREAD

**Discussion by mfarhan-nr** [containers/podman#24857] [Ideas]
I have scenario where, I am creating a Pod in Podman using the `--add-host`. This would add my Host machines Internal IP address to the `/etc/hosts` of all the containers that are part of the pod. And the same is reflected as well.
To be more precise, I have an RHEL instance, in which I have a service running on a tcp port say (8080). I am setting up a Pod using the `add-host` . I am starting an Ubuntu container which is added to the same Pod.
Now, in the ubuntu container's `/etc/hosts` I do see my host i.e, the RHEL's Internal IP address. But when I try to curl to the service in the host, I keep getting this error. `curl: (7) Failed to connect to <Host_IP_Address> port 8080 after 0 ms: Couldn't connect to server`
Things to call out:
1. The TCP 8080 port in Host Machine allows request from all. (i.e, 0.0.0.0/8080, localhost:8000, Host_IP_address:8000).
2. I don't see the entry of host.containers.internal. (I assumed it gets added for all containers).
3. I am running rootless Podman.
4. Version:      5.2.2

**Comment by sbrivio-rh** [comment]
Would it be possible for you to upgrade to Podman 5.3? Assuming you're talking about rootless containers, see: https://blog.podman.io/2024/10/podman-5-3-changes-for-improved-networking-experience-with-pasta/.
